### DISCUSSION TITLE
Running different checks at different frequencies

### DISCUSSION THREAD

**Discussion by Jul000** [spatie/laravel-health#65] [Q&A]
Hello,

How can we run some checks every minutes and other at a different intervals time ? like every 30 minutes ?

Thanks for your help.

Julien

**Comment by moisish** [comment]
On acustom check I have, I needed to run it on different internvals so I end up override the `new` method

```
    protected int $maxAgeInMinutes = 120;

    public static function new(): static
    {
        $instance = new static();

        $instance->spliceIntoPosition(1, '*/'.$instance->maxAgeInMinutes);

        return $instance;
    }
```

@freekmurze is this something you will consider a PR to add a method like setInterval or something?

**Reply by jaspertey** [reply]
~~I know this is from July 2022, but would you consider just going ahead and submitting a PR for what you proposed above?~~

EDIT: Upon closer inspection, the base abstract class `Spatie\Health\Checks\Check` that the health checks extend already uses the `Illuminate\Console\Scheduling\ManagesFrequencies` trait. If I understand correctly, this means all the standard scheduling helpers such as `->everyTenMinutes()`, `->everyThirtyMinutes()`, `->hourlyAt()` are available.

Furthermore, they would be available to use when registering the health check itself without having to override and bake it into the new() method.

e.g.,

```php
Health::checks([
    SecurityAdvisoriesCheck::new()->everyFifteenMinutes(),
]);
```
